collectors/build/jenkins-audit/src/main/java/com/capitalone/dashboard/util/AuditDataCSVFileCreator.java

```diff
@@ -0,0 +1,275 @@
+        this.dashboardAuditService = dashboardAuditService;
+        this.cmdbRepository = cmdbRepository;
+        createCSVFile();
+    }
+
+    private void createCSVFile() {
+        // This findByTimestampAfter can be changed based on the need.
+        //Iterable<Dashboard> recentDashboards = dashboardRepository.findByTimestampAfter(BEGIN_DATE);
+        Iterable<Dashboard> recentDashboards = dashboardRepository.findByTitle("CI304346");
+        AUDIT_TYPE.add(AuditType.ALL);
+        this.getAuditResults(recentDashboards, BEGIN_DATE);
+        if (!(entireCSVData == null && entireCSVData.isEmpty())) {
+            this.writeCSV();
+        }
+    }
+
+    private void writeCSV() {
+        try {
+            LOGGER.info("Writing CSV Data...");
+            Files.write(Paths.get(CSV_OUTPUT_FILE_PATH), String.join("\n", entireCSVData).getBytes());
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    private void getAuditResults(Iterable<Dashboard> dashboards, long timestamp) {
+        List<AuditResult> auditResults = new ArrayList();
+        // CSV Header
+        entireCSVData.add(String.join(",", Arrays.asList(CSV_HEADER)));
+
+        dashboards.forEach(dashboard -> {
+            try {
+                DashboardReviewResponse dashboardReviewResponse = dashboardAuditService.getDashboardReviewResponse(
+                        dashboard.getTitle(), dashboard.getType(), dashboard.getConfigurationItemBusServName(),
+                        dashboard.getConfigurationItemBusAppName(), BEGIN_DATE, END_DATE, AUDIT_TYPE
+                );
+                AuditResult auditResult = new AuditResult(dashboard.getId(), dashboardReviewResponse, timestamp);
+
+                auditResults.add(auditResult);
+                this.assignAuditDataValues(auditResult);
+                this.addCSVRowData(dashboard);
+
+            } catch (Exception e) {
+                e.printStackTrace();
+                LOGGER.error("AUDIT API ERROR - " + e.getMessage());
+                Cmdb cmdb = cmdbRepository.findByConfigurationItem(dashboard.getConfigurationItemBusServName());
+                List<String> csvAuditAPIFailedRow = Arrays.asList(cmdb.getOwnerDept(), dashboard.getConfigurationItemBusServName(),
+                        dashboard.getConfigurationItemBusServName(), cmdb.getBusinessOwner(), cmdb.getAppServiceOwner(),
+                        "", "NO_DATA", "NA", "", "AUDIT API ERROR - " + e.getMessage());
+                entireCSVData.add(String.join(",", csvAuditAPIFailedRow));
+            }
+        });
+    }
+
+    private void addCSVRowData(Dashboard dashboard) {
+        Cmdb cmdb = cmdbRepository.findByConfigurationItem(dashboard.getConfigurationItemBusServName());
+        lob = cmdb.getOwnerDept();
+        asv = dashboard.getConfigurationItemBusServName();
+        bap = dashboard.getConfigurationItemBusServName();
```

Comment:
These are internal terms, we need to rename these.

Reply:
internal terms are updated to have the generic names